src/test/java/com/teragrep/cfe_39/CombinedFullTest.java

```diff
@@ -0,0 +1,309 @@
+        if(!fs.exists(newDirectoryPath)) {
+            // Create new Directory
+            fs.mkdirs(newDirectoryPath);
+            LOGGER.info("Path {} created.", path);
+        }
+
+        // Use either HDFS-file modification timestamps or avro-mapred for pruning.
+
+        // The records are in this AVRO format:
+        // {"timestamp": 1650872092240000, "message": "25.04.2022 07:34:52.240 [WARN] com.teragrep.jla_02.Log4j2 [instanceId=01, thread=Thread-0, userId=, sessionId=, requestId=, SUBJECT=, VERB=, OBJECT=, OUTCOME=, message=Log4j2 warn audit says hi!]", "directory": "jla02logger", "stream": "test:jla02logger:0", "host": "jla-02.default", "input": "imrelp:cfe-06-0.cfe-06.default:", "partition": "8", "offset": 8, "origin": "jla-02.default"}
+        // Query handler must be implemented in a way that the AVRO files are first opened, then processed to syslog format and then sent to the query requester. The records are processed/filtered based on the given query conditions using MapReduce to make the code capable of processing the vast amounts of records that are expected.
+        // MapReduce functionalities of the Hadoop cluster: https://hadoop.apache.org/docs/stable/hadoop-mapreduce-client/hadoop-mapreduce-client-core/MapReduceTutorial.html
+        // Avro side of documentations for MapReduce: https://avro.apache.org/docs/1.11.1/mapreduce-guide/
+
+        // Another method for pruning aside using avro-mapred is to use modification timestamp of the avro-file stored in HDFS:
+        // fs.setTimes(new Path(path+"/"+0.8), Long.parseUnsignedLong("1675930598000"), -1);
+        // where mtime is modification time and atime is access time. -1 as input parameter leaves the original atime/mtime value as is.
+        FileStatus[] fileStatuses = fs.listStatus(new Path(newDirectoryPath + "/"));
+        long count = Arrays.stream(fileStatuses).count();
+        if (count != 0) {
+            if (config.getPruneOffset() != 157784760000L) {
+                Assertions.fail("There are files available in the database when there should be none.");
+            }
+            boolean delete = false;
+            for (FileStatus a : fileStatuses) {
+                // Delete old files
+                if (a.getModificationTime() < System.currentTimeMillis()) {
+                    delete = fs.delete(a.getPath(), true);
+                    Assertions.assertTrue(delete);
+                    LOGGER.info("Deleted file {}", a.getPath());
+                }
+            }
+            Assertions.assertTrue(delete);
+            LOGGER.info("All files were pruned properly.");
+        }else {
+            if (config.getPruneOffset() == 157784760000L) {
+                Assertions.fail("There were no files available in the database when there should be.");
+            }
+            LOGGER.info("No files available as they were pruned properly already!");
+        }
+
+        fs.close();
+    }
+
+    public void hdfsReadCheck() throws IOException {
```

Comment:
Tests should not throw nor try/catch anything, use `Assertions.assertDoesNotThrow`, `Assertions.assertThrows`, `Assertions.assertAll` and such instead

Reply:
Solved from commit 65342c2e865912f981dfe82f470cab6884f72cdb onward